opm/common/data/Wells.cpp

```diff
@@ -0,0 +1,70 @@
+/*
+  Copyright 2016 Statoil ASA.
+
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <opm/common/data/Wells.hpp> 
+
+namespace Opm {
+namespace data {
+
+    production::production( double w, double o, double g ) :
+        wat( w ),
+        oil( o ),
+        gas( g )
+    {}
+
+    production::production( const std::vector< double >& vec, size_t index,
+                            size_t wat_offset,
+                            size_t oil_offset,
+                            size_t gas_offset ) :
+        wat( vec[ index + wat_offset ] ),
+        oil( vec[ index + oil_offset ] ),
+        gas( vec[ index + gas_offset ] )
+    {}
+
+    completion::completion( size_t i, production p ) :
+        global_index( i ),
+        prod( p )
+    {}
+
+    Well::Well( production x, double bhp, std::map< size_t, completion >&& c ) :
+        prod( x ),
+        bhp( bhp ),
+        completions( std::move( c ) )
+    {}
+
+    const completion& Well::at( size_t i ) const {
```

Comment:
I'm not completely sold on why `index` on that line (which is just a convenience forward to `at`) is all that much more describing, but either way I'm not strongly opposed.

The constructors aren't even strictly necessary; rather, they're there (in the case of `completion` and `construction` so that `{}`-syntax picks up on more cases.